lib/one_model.ex

```diff
@@ -154,60 +154,98 @@ defmodule OneModel do
-      Get a list of #{@schema},s given a list of {field, value} pairs.
+      Get a list of #{@schema}s given a list of {field, value} pairs.
 
       ## Preload
 
       Pass a list of preloads with the `:preload` key.
 
+      ## Order by
+
+      Pass an atom or list of fields to sort by with the `:order_by` key.
+      Defaults to sorting by `:inserted_at` in ascending order.
+
+      ## Select
+
+      Pass an atom or list of fields to be selected with the `:select` key.
+      Passing an atom will return the value of the given atom key.
+      Passing a list of keys will return a map with the {key, value} pairs.
+
+      ## Limit
+
+      Pass an integer with the `:limit` key to specify the number of entries returned.
+
       ## Examples
 
-          #{@schema}.list_by(field1: value1, field2: field2, preload: [:association])
+          #{@schema}.list_by(field1: value1, field2: value2, preload: [:association])
+          #{@schema}.list_by(field1: value1, order_by: :field2)
+          #{@schema}.list_by(field1: value1, order_by: [asc: :field3])
+          #{@schema}.list_by(field1: value1, select: :field2)
+          #{@schema}.list_by(field2: value2, select: [:field1, field3], limit: 2)
       """
       @spec list_by(keyword()) :: [t()]
       def list_by(opts) do
-        opts
+        (opts ++ [order_by: :inserted_at])
         |> list_by_query()
-        |> order_by(asc: :inserted_at)
         |> @repo.all()
       end
 
       @doc """
       Build the list_by query.
       """
-      @spec list_by_query(keyword(), keyword()) :: Ecto.Query.t()
-      def list_by_query(params, opts \\ []) do
-        {preload, params} = Keyword.pop(params, :preload, [])
-        preload = if opts[:preload] == false or preload == [], do: false, else: preload
+      @spec list_by_query(keyword()) :: Ecto.Query.t()
+      def list_by_query(opts) do
+        {expressions, opts} = Keyword.split(opts, ~w(select order_by preload limit)a)
 
-        params
+        opts
         |> Enum.reduce(@schema, fn {k, v}, query ->
           where(query, [b], field(b, ^k) == ^v)
         end)
-        |> do_preload(preload)
+        |> do_order(expressions[:order_by])
+        |> do_preload(expressions[:preload])
+        |> do_select(expressions[:select])
+        |> do_limit(expressions[:limit])
       end
 
+      defp do_order(query, nil), do: query
+      defp do_order(query, false), do: query
+      defp do_order(query, []), do: query
+      defp do_order(query, order), do: order_by(query, ^order)
+
+      defp do_preload(query, nil), do: query
       defp do_preload(query, false), do: query
       defp do_preload(query, []), do: query
       defp do_preload(query, preload), do: preload(query, ^preload)
 
+      defp do_select(query, nil), do: query
+      defp do_select(query, []), do: query
+      defp do_select(query, key) when is_atom(key), do: select(query, [b], field(b, ^key))
+      defp do_select(query, [key]), do: select(query, [b], field(b, ^key))
```

Comment:
Lets keep a list of atoms consistanent with Ecto's API. i.e returns the schema with given keys fetch from the database and populated in the struct with remainig field defaults.